bin/filter_gtf.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import logging
+from itertools import groupby
+import argparse
+import re
+
+# Create a logger
+logging.basicConfig(format="%(name)s - %(asctime)s %(levelname)s: %(message)s")
+logger = logging.getLogger(__file__)
+logger.setLevel(logging.INFO)
+
+
+def is_header(line: str) -> bool:
+    """Returns True if the given line is a header line in a FASTA file."""
+    return line[0] == ">"
+
+
+def extract_fasta_seq_names(fasta_name: str) -> set:
+    """Extracts the sequence names from a FASTA file.
+
+    modified from Brent Pedersen
+    Correct Way To Parse A Fasta File In Python
+    given a fasta file. yield tuples of header, sequence
+    from https://www.biostars.org/p/710/
+
+    Args:
+      fasta_name: The path to the FASTA file.
+
+    Returns:
+      A set of the sequence names in the FASTA file.
+    """
+
+    # first open the file outside
+    fh = open(fasta_name)
+
+    # ditch the boolean (x[0]) and just keep the header or sequence since
+    # we know they alternate.
+    faiter = (x[1] for x in groupby(fh, is_header))
+
+    for i, header in enumerate(faiter):
+        line = next(header)
+        if is_header(line):
+            # drop the ">"
+            headerStr = line[1:].strip().split()[0]
+        yield headerStr
+
+
+def extract_genes_in_genome(fasta: str, gtf_in: str, gtf_out: str) -> None:
+    """Extracts the genes in the genome from a GTF file.
+
+    Args:
+      fasta: The path to the FASTA file.
+      gtf_in: The path to the input GTF file.
+      gtf_out: The path to the output GTF file.
+
+    Raises:
+      ValueError: If no overlap is found or if the GTF file is not tab delimited.
+    """
+
+    def is_tab_delimited(file):
+        with open(file, "r") as f:
+            return "\t" in f.readline()
+
+    if not is_tab_delimited(gtf_in):
+        raise ValueError("The GTF file is not tab delimited.")
```

Comment:
If we return the number of separators, we can reuse this info later.
```suggestion
    num_sep = tab_delimited(gtf_in)
    if num_sep != 8:
        raise ValueError("No valid tab-delimited GTF file.")
```